includes/payment-methods/class-upe-payment-gateway.php

```diff
@@ -1071,8 +1094,12 @@ public function log_payment_error_ajax() {
 			}
 			$order->add_order_note( $order_note );
 
+			self::remove_upe_payment_intent_from_session();
```

Comment:
It would be cleaner if instead this method is called in a hook rather than directly in the "log" method.

I tried this:

```php
add_action( 'wc_ajax_wcpay_log_payment_error', [ 'WCPay\Payment_Methods\UPE_Payment_Gateway', 'remove_upe_payment_intent_from_session' ] );
```

But that apparently didn't work as `remove_upe_payment_intent_from_session` was not called on client-side errors.